src/main/kotlin/com/cognifide/gradle/aem/instance/tasks/Create.kt

```diff
@@ -34,45 +26,71 @@ open class Create : Instance() {
 
         logger.info("Creating instances: ${uncreatedInstances.names}")
 
-        val backupZip = when (options.source) {
+        val backupZip = when (instanceOptions.source) {
             LocalInstanceOptions.Source.AUTO -> findRecentBackup()
             LocalInstanceOptions.Source.BACKUP_INTERNAL -> getInternalBackup()
             LocalInstanceOptions.Source.BACKUP_EXTERNAL -> getExternalBackup()
             LocalInstanceOptions.Source.NONE -> null
         }
         if (backupZip != null) {
-            val instanceRoot = File(aem.config.instanceRoot)
-
-            aem.logger.info("Extracting files from backup ZIP '$backupZip' to directory '$instanceRoot'")
-            aem.progressIndicator {
-                message = "Extracting backup ZIP: ${backupZip.name}, size: ${Formats.size(backupZip)}"
-                FileOperations.zipUnpack(backupZip, instanceRoot)
-            }
+            createFromBackup(uncreatedInstances, backupZip)
         } else {
-            if (options.jar == null || options.license == null) {
-                throw InstanceException("Cannot create instances due to lacking source files. " +
-                        "Ensure having specified: local instance ZIP url or jar & license url.")
-            }
+            createFromScratch(uncreatedInstances)
+        }
+
+        val createdInstances = uncreatedInstances.filter { it.created }
+
+        aem.notifier.notify("Instance(s) created", "Which: ${createdInstances.names}")
+    }
+
+    private fun createFromBackup(uncreatedInstances: List<LocalInstance>, backupZip: File) {
+        if (instances != uncreatedInstances) {
```

Comment:
This comparison looks weird, whats the rationale?

Reply:
we have defined instances and already created instances; previously we checked that uncreatedinstanced are not empty;

so now we need to be sure that count and names (author publish publish2) have not been changed. because of .equals impl of localinstance I can write this if looking extra simple :)